src/Elastic.Apm/DistributedTracing/TraceParent.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.DistributedTracing
+{
+	/// <summary>
+	/// This is an implementation of the
+	/// "https://www.w3.org/TR/trace-context/#traceparent-field" w3c 'traceparent' header draft.
+	/// elastic-apm-traceparent: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
+	/// (______________________)  () (______________________________) (______________) ()
+	///      v                     v                 v                        v         v
+	///  Header name           Version           Trace-Id                Span-Id     Flags
+	/// Since the w3c document is just a draft at the moment,
+	/// we don't use the official header name but prepend the custom prefix "Elastic-Apm-".
+	/// </summary>
+	internal struct TraceParent
+	{
+		private const int TraceParentLength = 55;
+		private const byte FlagRecorded = 1; // 00000001
+
+		internal const string TraceParentHeaderName = "elastic-apm-traceparent";
+
+		public static bool ValidateTraceParentValue(string traceParentValue, IApmLogger logger)
+		{
+			if (traceParentValue.Length != 55)
```

Comment:
Future versions of traceparent might become _longer_ than 55. The spec currently says an impl SHOULD try to parse a future version which might be longer than 55. The existing traceparent parts are guaranteed to stay the same. 

Of course it's just a "should".

see https://www.w3.org/TR/trace-context/#versioning-of-traceparent

Reply:
Yes, there was even a test for this in the test-suite `test_traceparent_version_0xcc`. I adapted the validation, we are ok with future version and best attempt as long as we can read `traceId`, `parentId` and the flags we are ok.